server/src/main/java/io/littlehorse/server/streams/topology/core/WfService.java

```diff
@@ -83,6 +85,25 @@ public WorkflowEventDefModel getWorkflowEventDef(WorkflowEventDefIdModel id) {
         return metadataManager.get(id);
     }
 
+    /**
+     *  Implementation inspired by {@link WfService#getWfSpec(String, Integer, Integer)}
+     */
+    public StructDefModel getStructDef(String name, Integer version) {
+        Supplier<StructDefModel> findStructDef = () -> {
+            final StructDefModel storedResult;
+
+            if (version != null) {
+                storedResult = metadataManager.get(new StructDefIdModel(name, version));
+            } else {
+                storedResult =
+                        metadataManager.getLastFromPrefix(StructDefIdModel.getPrefix(name), StructDefModel.class);
```

Comment:
I feel young again reading this code. It's been two years since I've touched the equivalents for `WfSpec`s and `TaskDef`s.